ninjabackend.py

```diff
@@ -407,19 +408,25 @@ def generate_po_install(self, d, elem):
                 d.po.append((src_file, self.environment.coredata.localedir, lang))
                 elem.add_dep(rel_src)
 
-    def generate_target_install(self, d):
-        libdir = self.environment.get_libdir()
-        bindir = self.environment.get_bindir()
+    def target_install_dir(self, t):
+        outdir = t.get_custom_install_dir()
+        if outdir is None:
+            if isinstance(t, build.Executable):
+                outdir = self.environment.get_bindir()
+            else:
+                outdir = self.environment.get_libdir()
+        return outdir
 
+    def generate_target_install(self, d):
         should_strip = self.environment.coredata.strip
         for t in self.build.get_targets().values():
             if t.should_install():
-                outdir = t.get_custom_install_dir()
-                if outdir is None:
-                    if isinstance(t, build.Executable):
-                        outdir = bindir
-                    else:
-                        outdir = libdir
+                outdir = self.target_install_dir(t)
+                if isinstance(t, build.Executable) and self.build.install_dependency_manifests:
+                    mf_src = self.get_target_filename(t) + '.dependencies.json'
```

Comment:
why not `'%s.dependencies.json' % self.get_target_filename(t)`? Let's use this for all string operations.